client/src/app/components/team/team.component.ts

```diff
@@ -1,66 +1,136 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { AppStateService } from 'src/app/shared/service/app-state.service';
-import { TeamControllerService, Team } from 'src/app/api';
+import { MatDialog, MatDialogConfig, MatSort, MatTableDataSource, MatPaginator } from '@angular/material';
+import { CreateTeamModalComponent } from 'src/app/shared/component/create-team-modal/create-team-modal.component';
+import { AddMemberModalComponent } from 'src/app/shared/component/add-member-modal/add-member-modal.component';
+import { TeamResourceService, Team, User } from 'src/app/api';
+import { TeamStateService } from 'src/app/shared/service/team-state.service';
+import { trigger, state, style, transition, animate } from '@angular/animations';
+import { DeleteTeamConfirmComponent } from 'src/app/shared/component/delete-team-confirm/delete-team-confirm.component';
+import { DeleteMemberConfirmComponent } from 'src/app/shared/component/delete-member-confirm/delete-member-confirm.component';
+import { DeleteLeaderConfirmComponent } from 'src/app/shared/component/delete-leader-confirm/delete-leader-confirm.component';
 
 @Component({
   selector: 'app-team',
   templateUrl: './team.component.html',
-  styleUrls: ['./team.component.css']
+  styleUrls: ['./team.component.css'],
+  animations: [
+    trigger('detailExpand', [
+      state('collapsed', style({height: '0px', minHeight: '0'})),
+      state('expanded', style({height: '*'})),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
+    ]),
+  ],
 })
+
 export class TeamComponent implements OnInit {
 
   teamName: string = '';
   companyName: string = '';
   team: any = '';
   createdTeam: Team;
-  teamNameToCreate: string = '';
-  companyNameToCreate: string = '';
   data: any;
   dataLeader: any;
   memberToAdd: string = '';
   teamNametoAdd: string = '';
   teamtoAddMember: Team;
 
-  isLeader: boolean = false;
   isSearchPressed: boolean = false;
+  teamList: Team[];
+  searchKey: string;
+  listData: MatTableDataSource<any>;
+  displayedColumns: string[] = ['name', 'company', 'actions'];
 
-  constructor(private teamService: TeamControllerService, private state: AppStateService) { }
+  expandedElement: any | null;
 
-  ngOnInit() {
+  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
 
+  constructor(
+    private teamService: TeamResourceService,
+    private dialog: MatDialog,
+    public state: AppStateService
+    ) { }
+
+  ngOnInit() {
+    this.getData();
   }
 
-  deleteTeam(teamName: string) {
-    this.isLeader = false;
-    this.teamService.deleteTeamUsingDELETE(teamName).subscribe(
-      (data: any) => {
+  getData() {
+    this.teamService.getAllTeamsUsingGET().subscribe(
```

Comment:
szintén, a hívás mehetne a teamStateService-be. A this.listData elkészítése pedig atkerulne a 

this.teamStateService.teams$.subscribe(
      //    IDE
)